File: cfgs/java_cfgs_foo/WAYS_TO_WRITE_N_AS_SUM_OF_TWO_OR_MORE_POSITIVE_INTEGERS.java
Nodes
1;special;;;
2;statement;int[] table = new int[foo_1 + 1];14;
3;statement;fun_1(foo_1, 0);15;
4;statement;foo_1[0] = 1;16;
5;statement;int i = 1;17;
6;conditional;foo_1 < foo_2;18;
7;statement;int j = foo_1;19;
8;conditional;foo_1 <= foo_2;20;
9;statement;foo_1[foo_2] += foo_3[foo_4 - foo_5];21;
10;statement;j++;21;
11;statement;i++;19;
12;statement;return foo_1[foo_2];22;
13;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;
5;7;12;
6;8;
7;9;11;
8;10;
9;8;
10;6;
11;13;
12;

File: results/transcoder/translations/py/WAYS_TO_WRITE_N_AS_SUM_OF_TWO_OR_MORE_POSITIVE_INTEGERS.cpp
int CountWays ( int n ) {
  int table [ n + 1 ] ;
  table [ 0 ] = 1 ;
  for ( int i = 1 ;
  i <= n ;
  i ++ ) {
    for ( int j = i ;
    j <= n ;
    j ++ ) {
      table [ j ] += table [ j - i ] ;
    }
  }
  return table [ n ] ;
}
